pkg/cli/validate_test.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+
+	jvspb "github.com/abcxyz/jvs/apis/v0"
+	"github.com/abcxyz/jvs/pkg/config"
+	"github.com/abcxyz/jvs/pkg/justification"
+	"github.com/abcxyz/pkg/testutil"
+	"github.com/google/go-cmp/cmp"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+func TestNewValidateCmd(t *testing.T) {
+	t.Parallel()
+
+	// Setup jwks server
+	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	keyID := "test_key_id"
+	ecdsaKey, err := jwk.FromRaw(privateKey.PublicKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := ecdsaKey.Set(jwk.KeyIDKey, keyID); err != nil {
+		t.Fatal(err)
+	}
+	jwks := make(map[string][]jwk.Key)
+	jwks["keys"] = []jwk.Key{ecdsaKey}
+
+	j, err := json.MarshalIndent(jwks, "", " ")
+	if err != nil {
+		t.Fatal("couldn't create jwks json")
+	}
+	path := "/.well-known/jwks"
+	mux := http.NewServeMux()
+	mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprintf(w, "%s", j)
+	})
+
+	svr := httptest.NewServer(mux)
+
+	t.Cleanup(func() {
+		svr.Close()
+	})
+
+	// Build test tokens
+	token := testTokenBuilder(t)
+	if err := jvspb.SetJustifications(token, []*jvspb.Justification{
+		{
+			Category: "explanation",
+			Value:    "test",
+		},
+		{
+			Category: "foo",
+			Value:    "bar",
+		},
+	}); err != nil {
+		t.Fatalf("failed to set justifications in token: %v", err)
+	}
+	signedToken := testSignToken(t, token, privateKey, keyID)
+	breakglassToken, err := jvspb.CreateBreakglassToken(token, "prod is down")
+	if err != nil {
+		t.Fatalf("failed to build breakglass token: %v", err)
+	}
+
+	cases := []struct {
+		name   string
+		config *config.CLIConfig
+		args   []string
+		pipe   bool
```

Comment:
Nit: `pipe` is generic and not really descriptive. Maybe `writeTokenToStdin` or have this as `stdin` and accept io.Reader, which drops the branching logic in the test.

```go
cases := []struct{}{
  // ...
  stdin io.Reader
}{
  {
    name: "from_stdin",
    // ...
    stdin: strings.NewReader(signedToken)
  },
}

for _, tc := range cases {
  tc := tc

  t.Run(tc.name, func(t *testing.T) {
    // ...

    stdout, stderr, gotErr = testExecuteCommandStdin(t, cmd, tc,stdin, tc.args...)
  })
}
```

Since `stdin` will default to `nil`, everything works.